downloader/downloader.go

```diff
@@ -568,6 +569,20 @@ func (downloader *Downloader) Download(data *extractors.Data) error {
 		return errors.Errorf("no stream named %s", streamName)
 	}
 
+	if downloader.option.AudioOnly {
+		for _, s := range sortedStreams {
+			// Looking for the best quality
+			matches, ok := regexp.MatchString("audio", s.Quality)
+			if ok != nil {
+				return errors.Errorf("No audio stream found")
```

Comment:
This error message is so misleading, I think we should just return the error returned by the `MatchString` function.

```suggestion
			if err != nil {
				return err
```